llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -6357,7 +6357,7 @@ OpenMPIRBuilder::InsertPointTy OpenMPIRBuilder::createTargetData(
   // Disable TargetData CodeGen on Device pass.
   if (Config.IsTargetDevice.value_or(false)) {
     if (BodyGenCB)
-      Builder.restoreIP(BodyGenCB(Builder.saveIP(), BodyGenTy::NoPriv));
+      Builder.restoreIP(BodyGenCB(CodeGenIP, BodyGenTy::NoPriv));
```

Comment:
Can you document this change in the PR description? Just looking at it I can't tell why it was wrong before and correct now.

Reply:
It's because `CodeGenIP` hasn't been restored by the `Builder` at this point. Instead of passing `CodeGenIP`, I've moved the `restoreIp` call upward.